rhp/v4/rhp.go

```diff
@@ -677,3 +727,30 @@ func NewRenewal(fc types.V2FileContract, prices HostPrices, rp RPCRenewContractP
 	}
 	return
 }
+
+// RefreshContract creates a new contract renewal from an existing contract revision
+func RefreshContract(fc types.V2FileContract, prices HostPrices, rp RPCRefreshContractParams) (renewal types.V2FileContractRenewal) {
+	// clear the old contract
+	renewal.FinalRevision = fc
+	renewal.FinalRevision.RevisionNumber = types.MaxRevisionNumber
+	renewal.FinalRevision.Filesize = 0
+	renewal.FinalRevision.FileMerkleRoot = types.Hash256{}
+	renewal.FinalRevision.RenterSignature = types.Signature{}
+	renewal.FinalRevision.HostSignature = types.Signature{}
```

Comment:
I don't think it's necessary to clear these fields; consensus should treat it as a valid resolution regardless.